src/MessageList/MessageListItem.vala

```diff
@@ -290,6 +291,85 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         });
     }
 
+    private Gtk.MenuButton actions_menu_button () {
+        var menu = new Gtk.Menu ();
+
+        var reply_item = new Gtk.MenuItem.with_label (_("Reply"));
+        reply_item.activate.connect (() => add_inline_composer (ComposerWidget.Type.REPLY));
+        menu.add (reply_item);
+
+        var reply_all_item = new Gtk.MenuItem.with_label (_("Reply to All"));
+        reply_all_item.activate.connect (() => add_inline_composer (ComposerWidget.Type.REPLY_ALL));
+        menu.add (reply_all_item);
+
+        var forward_item = new Gtk.MenuItem.with_label (_("Forward"));
+        forward_item.activate.connect (() => add_inline_composer (ComposerWidget.Type.FORWARD));
+        menu.add (forward_item);
+
+        menu.add (new Gtk.SeparatorMenuItem ());
+
+        var print_item = new Gtk.MenuItem.with_label (_("Print…"));
+        print_item.activate.connect (on_print);
+        menu.add (print_item);
+
+        menu.show_all ();
+
+        var menu_button = new Gtk.MenuButton ();
+        menu_button.image = new Gtk.Image.from_icon_name ("view-more-symbolic", Gtk.IconSize.MENU);
+        menu_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+        menu_button.tooltip_text = _("More");
+        menu_button.margin_top = 6;
+        menu_button.valign = Gtk.Align.START;
+        menu_button.halign = Gtk.Align.END;
+        menu_button.popup = menu;
+        /* Override default handler to stop event propagation. Otherwise clicking the menu will
+           expand or collapse the MessageListItem. */
+        menu_button.button_release_event.connect ((event) => {
+            menu_button.set_active (true);
+            return Gdk.EVENT_STOP;
+        });
+
+        menu_button.show ();
+        return menu_button;
+    }
+
+    private void add_inline_composer (ComposerWidget.Type composer_type) {
+        var message_list_box = (MessageListBox) get_parent();
```

Comment:
It looks like Mail 1.0.8 uses an approach where `MessageListItem` has a signal for reply, which chains to a signal in `MessageListBox`, which ultimately has a handler in `MainWindow`. That might be cleaner, and solve the problem of how to implement the hack where we scroll the inline composer into view.